eth/miner/miner.go

```diff
@@ -65,6 +65,7 @@ type Miner interface {
 }
 
 // miner implements the Miner interface.
+// TODO: RENAME TO STATE PROCESSOR OR DEPRECATE.
```

Comment:
deprecate would be fire but not convinced that we are setting everything up correct with just the geth one

Reply:
lifecycle management in general needs to be improved, later pr tho tbh